contracts/cw-native-staked-balance-voting/src/contract.rs

```diff
@@ -0,0 +1,299 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    coins, to_binary, BankMsg, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Response,
+    StdResult, Uint128,
+};
+use cw2::set_contract_version;
+use cw_controllers::ClaimsResponse;
+use cw_core_interface::voting::{TotalPowerAtHeightResponse, VotingPowerAtHeightResponse};
+use cw_utils::{must_pay, Duration};
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, GetConfigResponse, InstantiateMsg, MigrateMsg, QueryMsg};
+use crate::state::{Config, CLAIMS, CONFIG, DAO, MAX_CLAIMS, STAKED_BALANCES, STAKED_TOTAL};
+
+const CONTRACT_NAME: &str = "crates.io:cw-native-staked-balance-voting";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    let owner = match msg.owner {
+        Some(owner) => Some(deps.api.addr_validate(&owner)?),
+        None => None,
+    };
+
+    let manager = match msg.manager {
+        Some(manager) => Some(deps.api.addr_validate(&manager)?),
+        None => None,
+    };
+
+    let config = Config {
+        owner,
+        manager,
+        denom: msg.denom,
+        unstaking_duration: msg.unstaking_duration,
```

Comment:
How would you feel about making it not allowed to set an unstaking duration of zero (time or blocks)? We currently allow that in cw20-stake which I think was a bit of a mistake TBH.

Will add this to cw721-stake as well if you think it's a reasonable idea. :)